src/strat_engine/blockdev.rs

```diff
@@ -101,22 +103,24 @@ impl BlockDev {
             };
 
             try!(bd.write_sigblock());
-            bds.push(bd);
+            bds.insert(bd.dev_uuid.clone(), bd);
         }
         Ok(bds)
     }
 
     /// Gets device and device sizes, and returns an error if devices
     /// cannot be used by Stratis.
-    fn dev_info(paths: &[&Path], force: bool) -> EngineResult<Vec<(Device, u64)>> {
+    fn dev_info(devices: &BTreeSet<Device>, force: bool) -> EngineResult<Vec<(PathBuf, u64)>> {
```

Comment:
I mean the function and its one use could probably be split up as a map and a fold and a simple ```dev_info``` function without a loop in it, probably.